src/commands/settings/scan-members.ts

```diff
@@ -0,0 +1,79 @@
+import { CommandInteraction } from 'discord.js';
+import { WriteBatch } from '@google-cloud/firestore';
+
+// TODO: GDPR/CCPA compliance (applies to entire bot, tbh)
+// TODO: Could use cleanup, intentionally leaving until theres a usecase for this
+
+/** Collects guild member info into batches and processes them into Firebase */
+export async function scanMembers(
+  firestore: FirebaseFirestore.Firestore,
+  interaction: CommandInteraction
+) {
+  //Replying immediately allows command to run for longer than 3 seconds.
+  await interaction.reply({
+    content: 'Beginning user import',
+    ephemeral: true
+  });
+
+  const guild = interaction.guild;
+  if (guild === null) {
+    throw new Error('Guild cannot be null');
+  }
+  const guildRef = firestore.collection('guilds').doc(guild.id);
+
+  const batchArray: WriteBatch[] = [];
+  batchArray.push(firestore.batch());
+  let operationCount = 0;
+  let batchIndex = 0;
+
+  // Setup Firestore batches into array
+  for (const member of guild.members.cache.values()) {
+    if (member.user.bot) {
+      // Skips any bot members
+      continue;
+    }
+
+    batchArray[batchIndex].set(
+      guildRef.collection('members').doc(member.id),
+      { name: member.displayName },
+      { merge: true }
+    );
+    operationCount++;
+
+    // Firestore batches have a limit of 500 operations
+    if (operationCount === 499) {
+      batchArray.push(firestore.batch());
+      batchIndex++;
+      operationCount = 0;
+    }
+  }
+
+  // Write batches to Firestore
+  let batchCount = 1;
+  for (const batch of batchArray) {
+    await batch
```

Comment:
why are we using then & await together? I assume since we are using async/await in this file, we want to prefer that pattern over chaining?

Reply:
because this code is old and bad. Not gonna fix it at the moment, whole thing needs to get redone eventually. Just keeping it around because it has solutions that might be useful in future when it is rewritten.

Currently none of the code in this file is reachable through slash commands.